torchao/quantization/quant_api.py

```diff
@@ -664,6 +672,164 @@ def _int8_dynamic_activation_int4_weight_transform(
     return module
 
 
+@dataclass
+class Int8DynamicActivationIntxWeightConfig(AOBaseConfig):
+    """
+    Configuration for dynamically quantizing activations to torch.int8 and weights to torch.intx, with 1 <= x <= 8.
+    More specifically, activations are dynamically quantized to 8-bits at a per-token granularity with scales/zeros.
+    Weights are quantized with scales and optionally zeros (controlled by weight_zero_point_domain) in a groupwise or
+    channelwise manner using the number of bits specified by weight_dtype.
+
+    args:
+        weight_dtype: The dtype to use for weight quantization.  Must be torch.intx, where 1 <= x <= 8.
+            torch.intx with x < 8 requires TORCH_VERSION_AT_LEAST_2_6
+        weight_granularity: The granularity to use for weight quantization.  Must be PerGroup or PerRow.
+        weight_zero_point_domain: The domain to use for weight quantization.
+            Must be ZeroPointDomain.INT (if quantized weights have zeros) or ZeroPointDomain.NONE (if quantized weights do not have zeros).
+        weight_mapping_type: The type of mapping to use for the weight quantization.
+            Must be one of MappingType.ASYMMETRIC or MappingType.SYMMETRIC.  MappingType.SYMMETRIC requires ZeroPointDomain.NONE
+        weight_scale_dtype: The dtype to use for the weight scale.
+        act_mapping_type: The type of mapping to use for the activation quantization.
+            Must be one of MappingType.ASYMMETRIC or MappingType.SYMMETRIC.
+        layout: The layout to use for the packed weight tensor:
+            - PackedLinearInt8DynamicActivationIntxWeightLayout: this layout is optimized for CPU performance.
+            - QDQLayout: this layout represents the quantization with Q/DQ quant primitives, and is intended for
+                export applications like ExecuTorch.
+    """
+
+    weight_dtype: torch.dtype = torch.int8
+    weight_granularity: Union[PerRow, PerGroup] = PerGroup(32)
```

Comment:
I feel we should just make the type here `Granularity` and throw an error for unsupported types, so we don't tie ourselves to specific granularity in the signature itself